src/Quizzer.API.DAL/Repository/Repository.cs

```diff
@@ -55,10 +59,18 @@ public class RepositoryApiBase<TEntity> : IRepository<TEntity>, IDisposable
 
         public Guid Insert(TEntity entity) => DbSet().Add(entity).Entity.Id;
 
-        public Guid? Update(TEntity entity) {
+        public Guid? Update(TEntity entity, string navigationPropertyPath = "") {
+            return Update(entity, new string[] { navigationPropertyPath });
+        }
+
+        public Guid? Update(TEntity entity, IEnumerable<string> navigationPropertyPaths) {
             if (!Exists(entity.Id)) return null;
 
-            DbSet().Attach(entity);
+            TEntity? existingEntity = GetById(entity.Id, navigationPropertyPaths);
+
+            _mapper.Map(entity, existingEntity);
+
+
```

Comment:
Correct me, if I am wrong, but what is the purpose of the navigation property injection? As a caller you pass navigation property which is then used when getting an entity present in database. This will preload the navigation properties from the existing entity, but then you map new information which you changed on the entity itself. During the update, the information of other entities should not be important at the moment. You should not change foreign properties or foreign entity when you updating entity. At least in CB in ICS, the change of navigation properties were prohibited and this implementation leads consumer of this repository to alter properties which does not belong to the entity itself. I hope this is understandable. Not sure about your motivation then...

Reply:
Yes, but I am missing point why you need to include navigation paths only internally for this function when you only returning guid?

Reply:
I thought because of AutoMapper null properties, but it should not mapper since it's destination anyway.
TBH I don't understand why they do it like this in CB for recipies. I think Repository should work the way it is, How about we leave this PR right now and come back to it if Repository test's will fail?

Reply:
Ok we will see then...